numpy/core/src/umath/ufunc_object.c

```diff
@@ -3268,6 +3266,27 @@ PyUFunc_GenericFunction(PyUFuncObject *ufunc,
     return retval;
 }
 
+
+/*UFUNC_API*/
+NPY_NO_EXPORT int
+PyUFunc_GenericFunction(PyUFuncObject *ufunc,
+        PyObject *args, PyObject *kwds, PyArrayObject **op)
+{
+    /* NumPy 1.19, 2020-01-24 */
+    if (DEPRECATE(
+            "PyUFunc_GenericFunction() C-API function is deprecated "
+            "and expected to be removed rapidly. If you are using it (i.e. see "
+            "this warning/error), please notify the NumPy developers. "
+            "As of now it is expected that any use case is served better by "
+            "the direct use of `PyObject_Call(ufunc, args, kwargs)`. "
+            "PyUFunc_GenericFunction function has slightly different "
+            "untested behaviour.") < 0) {
+        return -1;
+    }
+    return PyUFunc_GenericFunction(ufunc, args, kwds, op);
```

Comment:
PyUFunc_GenericFunction_int

Reply:
Would have liked to claim it was on purpose to proof that really nobody uses it ;). Waiting for mailing list replies, but my guess is, nobody will really care (at least if I read your reply as "OK with some reservations", Chuck).